rls/src/test/java/io/grpc/rls/CachingRlsLbClientTest.java

```diff
@@ -636,6 +669,61 @@ private void setState(ChildPolicyWrapper policyWrapper, ConnectivityState newSta
     policyWrapper.getHelper().updateBalancingState(newState, policyWrapper.getPicker());
   }
 
+  @Test
+  public void metricGauges() throws ExecutionException, InterruptedException, TimeoutException {
+    setUpRlsLbClient();
+
+    verify(mockMetricRecorder).registerBatchCallback(cacheEntriesBatchCallbackCaptor.capture(),
+        argThat(new LongGaugeInstrumentArgumentMatcher("grpc.lb.rls.cache_entries")));
+    BatchCallback cacheEntriesBatchCallback = cacheEntriesBatchCallbackCaptor.getValue();
+    verify(mockMetricRecorder).registerBatchCallback(cacheSizeBatchCallbackCaptor.capture(),
+        argThat(new LongGaugeInstrumentArgumentMatcher("grpc.lb.rls.cache_size")));
+    BatchCallback cacheSizeBatchCallback = cacheSizeBatchCallbackCaptor.getValue();
+
+    // Verify the correct cache entries gauge value if requested at this point.
+    InOrder inOrder = inOrder(mockBatchRecorder);
+    cacheEntriesBatchCallback.accept(mockBatchRecorder);
+    inOrder.verify(mockBatchRecorder).recordLongGauge(
+        argThat(new LongGaugeInstrumentArgumentMatcher("grpc.lb.rls.cache_entries")), eq(0L),
+        isA(List.class), isA(List.class));
+
+    // Verify the correct cache size gauge value if requested at this point.
+    cacheSizeBatchCallback.accept(mockBatchRecorder);
```

Comment:
Looks like the second check here and at the end of the test should have been for `cache_size`.

Reply:
I read it wrong.